File: arbolBinario/.vscode/ArbolBinario.java
import java.util.Scanner;

public class ArbolBinario {

    static class Nodo {
        int valor;
        Nodo izq, der;
        Nodo(int v) { valor = v; }
    }

    static Nodo insertar(Nodo raiz, int v) {
        if (raiz == null) return new Nodo(v);
        if (v <= raiz.valor) raiz.izq = insertar(raiz.izq, v);
        else                 raiz.der = insertar(raiz.der, v);
        return raiz;
    }

    static void preorden(Nodo r) {
        if (r == null) return;
        System.out.print(r.valor + " ");
        preorden(r.izq);
        preorden(r.der);
    }

    static void inorden(Nodo r) {
        if (r == null) return;
        inorden(r.izq);
        System.out.print(r.valor + " ");
        inorden(r.der);
    }

    static void postorden(Nodo r) {
        if (r == null) return;
        postorden(r.izq);
        postorden(r.der);
        System.out.print(r.valor + " ");
    }

    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        System.out.println("Inserta enteros para el arbol (termina con -1):");

        Nodo raiz = null;

        while (true) {
            if (!sc.hasNextInt()) {
                System.err.println("Entrada no valida.");
                return;
            }
            int x = sc.nextInt();
            if (x == -1) break;
            raiz = insertar(raiz, x);
        }

        System.out.print("\nRecorrido PREORDEN:  ");
        preorden(raiz);
        System.out.print("\nRecorrido INORDEN:   ");
        inorden(raiz);
        System.out.print("\nRecorrido POSTORDEN: ");
        postorden(raiz);
        System.out.println();
    }
}

File: arbolBinario/.vscode/ArbolBinario.f90
program ArbolBinario
    implicit none

    type :: Nodo
        integer :: valor = 0
        type(Nodo), pointer :: izq => null()
        type(Nodo), pointer :: der => null()
    end type Nodo

    type(Nodo), pointer :: raiz => null()
    integer :: x, ios
    character(len=1) :: pausa

    print *, 'Inserta enteros para el arbol (termina con -1):'

    do
        read(*,*, iostat=ios) x
        if (ios /= 0) then
            print *, 'Entrada no valida.'
            call liberar(raiz)
            stop 1
        end if
        if (x == -1) exit
        call insertar(raiz, x)
    end do

    write(*,'(A)', advance='no') new_line('a')//'Recorrido PREORDEN:  '
    call preorden(raiz)
    print *
    write(*,'(A)', advance='no') 'Recorrido INORDEN:   '
    call inorden(raiz)
    print *
    write(*,'(A)', advance='no') 'Recorrido POSTORDEN: '
    call postorden(raiz)
    print *

    call liberar(raiz)

    ! 👇 "Pausa" antes de salir
    print *, new_line('a'), 'Presiona ENTER para salir...'
    read(*,'(A)') pausa

contains

    recursive subroutine insertar(raiz, v)
        type(Nodo), pointer :: raiz
        integer, intent(in) :: v
        if (.not. associated(raiz)) then
            allocate(raiz)
            raiz%valor = v
            raiz%izq => null()
            raiz%der => null()
        else if (v <= raiz%valor) then
            call insertar(raiz%izq, v)
        else
            call insertar(raiz%der, v)
        end if
    end subroutine insertar

    recursive subroutine preorden(r)
        type(Nodo), pointer, intent(in) :: r
        if (.not. associated(r)) return
        write(*,'(I0,1X)', advance='no') r%valor
        call preorden(r%izq)
        call preorden(r%der)
    end subroutine preorden

    recursive subroutine inorden(r)
        type(Nodo), pointer, intent(in) :: r
        if (.not. associated(r)) return
        call inorden(r%izq)
        write(*,'(I0,1X)', advance='no') r%valor
        call inorden(r%der)
    end subroutine inorden

    recursive subroutine postorden(r)
        type(Nodo), pointer, intent(in) :: r
        if (.not. associated(r)) return
        call postorden(r%izq)
        call postorden(r%der)
        write(*,'(I0,1X)', advance='no') r%valor
    end subroutine postorden

    recursive subroutine liberar(r)
        type(Nodo), pointer :: r
        if (.not. associated(r)) return
        call liberar(r%izq)
        call liberar(r%der)
        deallocate(r)
        nullify(r)
    end subroutine liberar

end program ArbolBinario
